src/Microsoft.AspNetCore.Mvc.TagHelpers/DependencyInjection/TagHelperExtensions.cs

```diff
@@ -26,13 +28,60 @@ public static IMvcCoreBuilder AddCacheTagHelper(this IMvcCoreBuilder builder)
                 throw new ArgumentNullException(nameof(builder));
             }
 
-            builder.Services.TryAddTransient<IDistributedCacheTagHelperStorage, DistributedCacheTagHelperStorage>();
-            builder.Services.TryAddTransient<IDistributedCacheTagHelperFormatter, DistributedCacheTagHelperFormatter>();
+            builder.Services.TryAddSingleton<IDistributedCacheTagHelperStorage, DistributedCacheTagHelperStorage>();
+            builder.Services.TryAddSingleton<IDistributedCacheTagHelperFormatter, DistributedCacheTagHelperFormatter>();
             builder.Services.TryAddSingleton<IDistributedCacheTagHelperService, DistributedCacheTagHelperService>();
 
             // Required default services for cache tag helpers
-            builder.Services.TryAddSingleton<IDistributedCache, MemoryDistributedCache>();
-            builder.Services.TryAddSingleton<IMemoryCache, MemoryCache>();
```

Comment:
Do we need an announcement for this since we could be breaking users consuming `IMemoryCache` in their applications?